pkg/ansible/proxy/proxy.go

```diff
@@ -87,6 +96,69 @@ func InjectOwnerReferenceHandler(h http.Handler) http.Handler {
 			log.V(1).Info("serialized body", "Body", string(newBody))
 			req.Body = ioutil.NopCloser(bytes.NewBuffer(newBody))
 			req.ContentLength = int64(len(newBody))
+		case http.MethodGet:
+			// GET request means we need to check the cache
+			rf := k8sRequest.RequestInfoFactory{APIPrefixes: sets.NewString("api", "apis"), GrouplessAPIPrefixes: sets.NewString("api")}
+			r, err := rf.NewRequestInfo(req)
+			if err != nil {
+				log.Error(err, "failed to convert request")
+				return
+			}
+
+			// check if resource is present on request
+			if !r.IsResourceRequest {
+				break
+			}
+
+			if strings.HasPrefix(r.Path, "/version") {
+				// Temporarily pass along to API server
+				// Ideally we cache this response as well
+				break
+			}
+
+			gvr := schema.GroupVersionResource{
+				Group:    r.APIGroup,
+				Version:  r.APIVersion,
+				Resource: r.Resource,
+			}
+			k, err := restMapper.KindFor(gvr)
+			if err != nil {
+				// break here in case resource doesn't exist in cache
+				log.Error(err, "didn't find kind")
+				break
+			}
+
+			un := unstructured.Unstructured{}
+			un.SetGroupVersionKind(k)
+			obj := client.ObjectKey{Namespace: r.Namespace, Name: r.Name}
+			err = informerCache.Get(context.Background(), obj, &un)
+			if err != nil {
+				// break here in case resource doesn't exist in cache but exists on APIserver
+				// This is very unlikely but provides user with expected 404
+				log.Error(err, "didn't find object in cache")
+				break
+			}
+
+			i := bytes.Buffer{}
+			resp, err := json.Marshal(un.Object)
+			if err != nil {
+				// return will give a 500
```

Comment:
How is that implemented?

Reply:
The proxy never writes a response and errors out processing the request so returning gives a 500. I believe this is the proper response when we have an error retrieving and writing objects from the cache.

Reply:
The other option is that we could panic with `ErrAbortHandler` which would also give a 500.

Reply:
I see. I agree with using 500. It just isn't obvious, to me anyway, that not writing a response will lead to a 500. Is that documented somewhere? I tried searching the go http docs for what behavior is expected in a handler function, but didn't find much.

The other option would be to call `http.Error()` explicitly. Are there pros and cons of doing that vs this?

Reply:
I like using `http.Error()` so it's more explicit. Thanks!